### PR TITLE
(CDAP-5242) Added documentation for audit publishing.

### PR DESCRIPTION
**Pull request by bdmogal**
(CDAP-5247) Deprecated Metadata updates notification feed.

Issues:
* [CDAP-5242](https://issues.cask.co/browse/CDAP-5242)
* [CDAP-5247](https://issues.cask.co/browse/CDAP-5247)

[Quick Build 8](http://builds.cask.co/browse/CDAP-DOB198-8)

Relevant doc pages:
* [Audit Logging](http://builds.cask.co/artifact/CDAP-DOB198/shared/build-8/Docs-HTML/3.4.0-SNAPSHOT/en/developers-manual/building-blocks/audit-logging.html)
* [Metadata Feed Deprecation](http://builds.cask.co/artifact/CDAP-DOB198/shared/build-8/Docs-HTML/3.4.0-SNAPSHOT/en/developers-manual/building-blocks/metadata-lineage.html#metadata-update-notifications)

### CODE REVIEW

File: cdap-docs/integrations/source/partners/cloudera/navigator-integration.rst [hunk 1]
```diff
@@ -33,9 +34,9 @@ To use the Navigator Integration App, you need CDAP version 3.3.0 (or higher) an
 
 Metadata Publishing to Kafka
 ----------------------------
-The Navigator Integration App contains a flow that subscribes to the Kafka topic to which the CDAP Metadata service publishes
-the metadata updates. Hence, before using this application, you should enable publishing of metadata updates to
-Kafka, as described in the CDAP document :ref:`Enable Metadata Update Notifications <metadata-update-notifications>`.
+The Navigator Integration App contains a flow that subscribes to the Kafka topic to which the CDAP audit logs are
+published. Hence, before using this application, you should
+:ref:`enable audit publishing <auditing-configuring-audit-publishing>`.
```
Comment (gokulavasan) [inline cdap-docs/integrations/source/partners/cloudera/navigator-integration.rst]: Can you also make the doc changes in the cdap-navigator repo? Thanks!

Comment (bdmogal) [inline cdap-docs/integrations/source/partners/cloudera/navigator-integration.rst]: Yup, will do.

File: cdap-docs/developers-manual/source/building-blocks/index.rst [hunk 2]
```diff
@@ -66,6 +66,10 @@ write data through the data abstraction layer in CDAP.
   Access of these entities is tracked, and you can view the :doc:`lineage <metadata-lineage>` of datasets and streams.
   With a lineage diagram, you can then drill down into the metadata of its nodes. 
 
+- :doc:`Auditing <auditing>` provides a chronological ledger containing evidence of operations or
+  changes on CDAP entities. This information can be used to capture a trail of the activities that 
+  determined the state of an entity at a given point in time.
```
Comment (bdmogal) [inline cdap-docs/developers-manual/source/building-blocks/index.rst]: determine

File: cdap-docs/developers-manual/source/building-blocks/metadata-lineage.rst [hunk 3]
```diff
@@ -102,11 +102,17 @@ This table lists the **system** metadata annotations of CDAP entities:
...
 =============================
+.. topic::  **Note: Metadata Update Notifications Deprecated**
+
+    As of *CDAP v3.4.0*, *Metadata Update Notifications* have been deprecated, pending removal in a later
+    version. The :ref:`CDAP Audit Notifications <auditing>` contain notifications for metadata changes. Please change
+    all uses of *Metadata Update Notifications* to consume only those messages from the audit feed that have the
+    ``type`` field set to ``METADATA_CHANGE``.
+
 CDAP has the capability of publishing notifications to an external Apache Kafka instance
 upon metadata updates.
```
Comment (awholegunch) [inline cdap-docs/developers-manual/source/building-blocks/metadata-lineage.rst]: The three properties mentioned below this line should be marked as deprecated in the ``cdap-default.xml``. The MD5 hash in the admin-manaul/build.sh will need updating to reflect that change.

Comment (awholegunch) [inline cdap-docs/developers-manual/source/building-blocks/metadata-lineage.rst]: Adding the notice to the release notes.

Comment (bdmogal) [inline cdap-docs/developers-manual/source/building-blocks/metadata-lineage.rst]: done. please see last commit.

File: cdap-common/src/main/resources/cdap-default.xml [hunk 4]
```diff
@@ -1033,7 +1033,7 @@
       Determines if metadata updates will be published to Apache Kafka.
       External systems can subscribe to the Kafka topic determined by
       ${metadata.updates.kafka.topic} to receive notifications of metadata
-      updates.
+      updates (deprecated)
```
Comment (awholegunch) [inline cdap-common/src/main/resources/cdap-default.xml]: This should end with a period:
```
updates (deprecated).

**Comment by awholegunch** [comment]
LGTM.

**Comment by bdmogal** [comment]
Thanks @awholegunch. Will wait for @poornachandra to sign off on the content.

File: cdap-docs/developers-manual/source/building-blocks/index.rst [hunk 5]
```diff
@@ -65,6 +66,10 @@ write data through the data abstraction layer in CDAP.
   Access of these entities is tracked, and you can view the :doc:`lineage <metadata-lineage>` of datasets and streams.
   With a lineage diagram, you can then drill down into the metadata of its nodes. 
 
+- :doc:`Auditing <auditing>` provides a chronological ledger containing evidence of operations or
```
Comment (poornachandra) [inline cdap-docs/developers-manual/source/building-blocks/index.rst]: Should we use Audit Log  instead of Auditing?

**Comment by bdmogal** [comment]
Thanks for the updates @awholegunch. @poornachandra can you take another look?

File: cdap-docs/developers-manual/source/building-blocks/index.rst [hunk 6]
```diff
@@ -65,6 +66,10 @@ write data through the data abstraction layer in CDAP.
   Access of these entities is tracked, and you can view the :doc:`lineage <metadata-lineage>` of datasets and streams.
   With a lineage diagram, you can then drill down into the metadata of its nodes. 
 
+- :doc:`Audit Logging <audit-logging>` provides a chronological ledger containing evidence of operations or
+  changes on CDAP entities. This information can be used to capture a trail of the activities that 
+  determined the state of an entity at a given point in time.
```
Comment (poornachandra) [inline cdap-docs/developers-manual/source/building-blocks/index.rst]: determine?

**Comment by poornachandra** [comment]
LGTM

**Pull request merged**